File: Rights.java
import java.util.HashMap;

public class Rights {
    private int id; // Індіфікатор рівня прав
    private String slug; // Найменування рівня прав
    private HashMap<String, String> perms = new HashMap<>(); // Словник усіх прав користувача // словник.put("книга", "book");

    public Rights(int id, String slug, HashMap<String, String> perms) {
        this.id = id;
        if (slug.matches("[a-zA-Z]+") && !slug.isEmpty())
            this.slug = slug;
        else
            throw new IllegalArgumentException("Rights конструктор: Назва рівня прав неповинна бути пустою і містити нелатинськи символи");
        if (perms != null)
            this.perms = perms;
        else
            throw new IllegalArgumentException("Rights конструктор: Словник повноважень не може бути рівним null");
    }

    public int getId() {
        // Повертає індіфікатор рівня прав
        return id;
    }

    public String getSlug() {
        // Повертає найменування рівня прав
        return slug;
    }
    public void setSlug(String slug) {
        // Змінює найменування рівня прав
        // Якщо слаг містить тільки латинські літери
        if (slug.matches("[a-zA-Z]+") && !slug.isEmpty()) // і довже 0 символів
            this.slug = slug; // змінює найменування рівня прав
        else // В противному випадку
            throw new IllegalArgumentException("setSlug(String slug) - Назва рівня прав неповинна бути пустою і містити нелатинськи символи");
    }

    public HashMap<String, String> getPerms() {
        // Повертає список усіх повноваженнь рівня прав
        return perms;
    }
    public void setPerms(HashMap<String, String> perms) {
        // Змінює список повноважень рівня прав
        if (perms != null)
            this.perms = perms;
        else
            throw new IllegalArgumentException("setPerms(HashMap<String, String> perms) - Словник повноважень не може бути рівним null");
    }
}

File: Rights.php
<?php

require_once 'Utils.php';
require_once 'User.php';

class Rights {
    private int $id; // Ідентифікатор рівня прав
    private string $slug; // Найменування рівня прав
    private array $perms; // Словник усіх прав користувача

    /**
     * @param int $id
     * @param string $slug
     * @param array $perms
     */
    public function __construct(int $id, string $slug, array $perms) {
        $this->id = $id;
        $this->validateSlug($slug); // Перевіряє, чи відповідає слаг нормам
        $this->slug = $slug;
        $this->perms = $perms;
    }

    public static function get(int $id): Rights {
        $rights = require 'settings/rights_list.php';
        return $rights[$id-1];
    }

    public function getId(): int {
        // Повертає Ідентифікатор рівня прав
        return $this->id;
    }

    public function getSlug(): string {
        // Повертає слаг замовлення
        return $this->slug;
    }
    public function setSlug(string $slug): void {
        // Встановлює слаг
        $this->slug = $slug;
    }
    private function validateSlug(string $slug): void {
        // Перевіряє, чи складається slug тільки з латинських літер, і чи містить заборонені символи
        Utils::validateName($slug, "validateUsername(string $slug)"); // Utils. Викликає помилку, якщо довжина строки = 0
        if (!preg_match('/^[a-zA-Z]+$/', $slug))
            throw new InvalidArgumentException('validateSlug(string $slug): Очікується, що slug буде містити тільки латинські літери');
    }

    public function getPerms(): array {
        // Повертає словник прав
        return $this->perms;
    }

    public function setPerms(array $perms): void {
        // Встановлює словник прав
        $this->perms = $perms;
    }


}